packages/dev/loaders/src/glTF/2.0/Extensions/KHR_animation_pointer.data.ts

```diff
@@ -46,251 +48,287 @@ function getTextureTransformTree(textureName: string) {
 
 class CameraAnimationPropertyInfo extends AnimationPropertyInfo {
     /** @internal */
-    public buildAnimations(target: ICamera, name: string, fps: number, keys: any[], callback: (babylonAnimatable: IAnimatable, babylonAnimation: Animation) => void): void {
-        callback(target._babylonCamera!, this._buildAnimation(name, fps, keys));
+    public buildAnimations(target: ICamera, name: string, fps: number, keys: any[]) {
+        return [{ babylonAnimatable: target._babylonCamera!, babylonAnimation: this._buildAnimation(name, fps, keys) }];
     }
 }
 
 class MaterialAnimationPropertyInfo extends AnimationPropertyInfo {
     /** @internal */
-    public buildAnimations(target: IMaterial, name: string, fps: number, keys: any[], callback: (babylonAnimatable: IAnimatable, babylonAnimation: Animation) => void): void {
-        for (const fillMode in target._data!) {
-            callback(target._data![fillMode].babylonMaterial, this._buildAnimation(name, fps, keys));
+    public buildAnimations(target: IMaterial | Material, name: string, fps: number, keys: any[]) {
+        const babylonAnimations: { babylonAnimatable: IAnimatable; babylonAnimation: Animation }[] = [];
+        if ((target as IMaterial)._data) {
+            const targetAsIMaterial = target as IMaterial;
+            for (const fillMode in targetAsIMaterial._data!) {
+                babylonAnimations.push({
+                    babylonAnimatable: targetAsIMaterial._data![fillMode].babylonMaterial,
+                    babylonAnimation: this._buildAnimation(name, fps, keys),
+                });
+            }
+        } else {
+            babylonAnimations.push({ babylonAnimatable: target as Material, babylonAnimation: this._buildAnimation(name, fps, keys) });
         }
+        return babylonAnimations;
     }

```

Comment:
We chatted offline about this one. It's a bit weird that we need both IMaterial and Material as types. You were going to investigate.

Reply:
This was done, it is now working the same as animation pointer (whereas I believe it is working correctly because there is only one fillMode)